rama-http/src/service/fs/serve_dir/mod.rs

```diff
@@ -515,7 +538,7 @@ impl<F> ServeDir<F> {
 
         let path_to_file = match self
             .variant
-            .build_and_validate_path(&self.base, req.uri().path())
+            .build_and_validate_path(self.base.clone(), req.uri().path())
```

Comment:
why all these `self.base.clone()` are needed all of a sudden? We should still be able to take by reference, no? What am I missing?